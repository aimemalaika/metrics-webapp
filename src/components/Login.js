import { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import logo from '../image/logo.png';

const Login = () => {
  const [username, setUsername] = useState('aimemalaika');
  const [password, setPassword] = useState('aimemalaika');
  const navigate = useNavigate();

  const auth = () => {
    if (username === 'aimemalaika' && password === 'aimemalaika') {
      toast.success('Logged in successfully!', {
        position: 'top-right',
        autoClose: 500,
        hideProgressBar: false,
        closeOnClick: true,
        pauseOnHover: true,
        draggable: true,
        progress: undefined,
      });
      setTimeout(() => { navigate('/countries'); }, 500);
    } else {
      toast.error('Username or password is incorect!', {
        position: 'top-right',
        autoClose: 500,
        hideProgressBar: false,
        closeOnClick: true,
        pauseOnHover: true,
        draggable: true,
        progress: undefined,
      });
    }
  };

  return (
    <div className="form-section">
      <ToastContainer
        position="top-right"
        autoClose={500}
        hideProgressBar={false}
        newestOnTop={false}
        closeOnClick
        rtl={false}
        pauseOnFocusLoss
        draggable
        pauseOnHover
      />
      <ToastContainer />
      <img src={logo} alt="logo" className="app-logo" />
      <form className="form-login">
        <p className="login-header">Login to access data</p>
        <input type="text" value={username} onChange={(e) => setUsername(e.target.value)} required placeholder="username" />
        <input type="password" value={password} onChange={(e) => setPassword(e.target.value)} required placeholder="password" />
        <button onClick={auth} type="button">Sign In</button>
      </form>
    </div>
  );
};

export default Login;
