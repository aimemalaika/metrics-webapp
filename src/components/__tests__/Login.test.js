import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Login from '../Login';

describe('Test the Country list', () => {
  it('Country section renders correctly', () => {
    const country = renderer.create(
      <MemoryRouter>
        <Login />
      </MemoryRouter>,
    ).toJSON();
    expect(country).toMatchSnapshot();
  });
});
