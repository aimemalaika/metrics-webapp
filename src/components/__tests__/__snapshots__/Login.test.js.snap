// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Country list Country section renders correctly 1`] = `
<div
  className="form-section"
>
  <div
    className="Toastify"
  />
  <div
    className="Toastify"
  />
  <img
    alt="logo"
    className="app-logo"
    src="logo.png"
  />
  <form
    className="form-login"
  >
    <p
      className="login-header"
    >
      Login to access data
    </p>
    <input
      onChange={[Function]}
      placeholder="username"
      required={true}
      type="text"
      value="aimemalaika"
    />
    <input
      onChange={[Function]}
      placeholder="password"
      required={true}
      type="password"
      value="aimemalaika"
    />
    <button
      onClick={[Function]}
      type="button"
    >
      Sign In
    </button>
  </form>
  <div
    className="animate-load"
  >
    <svg
      className="spinner"
      fill="currentColor"
      height="1em"
      icon="spinner"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
      />
    </svg>
    Loading ...
  </div>
</div>
`;
